st2common/st2common/router.py

```diff
@@ -499,3 +499,16 @@ def as_wsgi(self, environ, start_response):
         req = Request(environ)
         resp = self(req)
         return resp(environ, start_response)
+
+    def _get_model_instance(self, model_cls, data):
+        try:
+            instance = model_cls(**data)
+        except TypeError as e:
+            # Throw a more user-friendly exception when input data is not an object
+            if 'type object argument after ** must be a mapping, not' in str(e):
+                msg = ('Input body needs to be an object, got: %s' % (type(data)))
```

Comment:
Strictly speaking, user provides JSON array, not <type 'list'>. He should not have to know python specifics.

Reply:
True, but I'm not sure it's worth keeping such a type map inside the code, especially since the input could also be YAML.

Reply:
ef93708d7969a49be5b4d6f362af0ec8670c7d4e